### ISSUE TITLE
pls not working

### ISSUE THREAD

**Issue by Axel-Erfurt** [Borim7/radio-tray#1]
Got ot working by changing _PlsPlaylistDecoder.py_ to

```python
##########################################################################
# Copyright 2009 Carlos Ribeiro
#
# This file is part of Radio Tray
#
# Radio Tray is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 1 of the License, or
# (at your option) any later version.
#
# Radio Tray is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with Radio Tray.  If not, see <http://www.gnu.org/licenses/>.
#
##########################################################################
#import urllib.request, urllib.error, urllib.parse
#from radiotray.lib.common import USER_AGENT
import logging
import requests

class PlsPlaylistDecoder:

    def __init__(self):
        self.log = logging.getLogger('radiotray')
        self.log.debug('PLS playlist decoder')
        

    def isStreamValid(self, contentType, firstBytes):

        if(('audio/x-scpls' in contentType) or ('application/pls+xml' in contentType) or (firstBytes.strip().lower().startswith(b'[playlist]'))):
            self.log.info('Stream is readable by PLS Playlist Decoder')
            return True
        else:
            return False



    def extractPlaylist(self,  url):
            
            self.log.info('Downloading playlist...')
            headers = {'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.95 Safari/537.36'}
            str = requests.get(url, headers=headers)
            
            self.log.info('Playlist downloaded')
            self.log.info('Decoding playlist...')
            
            playlist = []
            lines = str.text.splitlines()
            for line in lines:

                if line.startswith("File") == True:

                        list = line.split("=", 1)
                        playlist.append(list[1])
      
            
            return playlist
```

**Comment by Borim7** [comment]
The suggestion to change to requests module looks good. Propably the best will be to replace urllib completely by requests module.

**Comment by Borim7** [comment]
fixed with 091864d
